File: 242. Valid Anagram/C++/Solution.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.length() != t.length()) {
            return false;
        }
        vector<int> counts(26, 0);
        for (auto item : s) {
            counts[item - 'a']++;
        }
        for (char item : t) {
            counts[item - 'a']--;
        }
        for (auto count : counts) {
            if (count != 0) {
                return false;
            }
        }
        return true;
    }
};

File: 242. Valid Anagram/Python/Solution.py
class Solution:
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) != len(t):
            return False
        counts = [0] * 26
        for item in s:
            counts[ord(item) - ord('a')] += 1
        for item in t:
            counts[ord(item) - ord('a')] -= 1
        return all(count == 0 for count in counts)
